src/Disks/ObjectStorages/ReadOnlyMetadataStorage.h

```diff
@@ -0,0 +1,165 @@
+#pragma once
+
+#include <Disks/ObjectStorages/IMetadataStorage.h>
+#include <Common/ErrorCodes.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int NOT_IMPLEMENTED;
+}
+
+class ReadOnlyMetadataStorage;
+
+/// Readonly, throws NOT_IMPLEMENTED error.
+/// Can be used to add limited read-only support of MergeTree.
+class ReadOnlyMetadataTransaction : public IMetadataTransaction
```

Comment:
Probably bad naming. "Read-only transaction" looks strange. I mean a transaction is always changing something in a batch, how can it be readonly?

Reply:
Yes, I thought about for awhile but did not managed to came up with a better alternative. Do you have some ideas?

P.S. I thought about using Limited/Surrogate/... but all of these also sound strange and does not clarify it a lot.

Reply:
Maybe `FakeMetadataTransactionForReadOnlyStorage` will be more clear, what do you think?

Reply:
a) it is toooo long
b) there is already `FakeMetadataStorageFromDisk` 
c) it does not give any benefit since you do not need transactions for readonly storage

Ok, let's just throw NOT_IMPLEMENTED from all methods.

Reply:
ok